File: Openearthtool/python/applications/KMLDap/kmldap/templates/code.py
import datetime
import pydap.client

from numpy import any, all, ma, apply_along_axis, nonzero, array, isnan, nan
from scipy.interpolate import interp1d
from functools import partial

url = '${config.get("jarkus.url")}'
# Create a dataset object.
ds = pydap.client.open_url(url)
transect_index = nonzero(ds['id'] == ${transect.id})[0][0]
#

# pydap returns altitude as a grid object, so we need to specify that we want only the values
z = ds['altitude']['altitude'][:,transect_index,:]
z[z==-9999] = nan
# read cross shore distance
cross_shore = ds['cross_shore'][:]
# read and convert time
time = ds['time'][:]
# you could convert these using: [datetime.timedelta(days=day) + datetime.date(1970,1,1) for day in ds['time'][:]]


# Fill missings to create a better plot
def fillmissing(x,y):
    """fill nans in y using linear interpolation"""
    f = interp1d(x[~isnan(y)], y[~isnan(y)], kind='linear',bounds_error=False, copy=True)
    new_y = f(list(x)) #some bug causes it not to work if x is passed directly
    return new_y
# define an interpolation for a row by partial function application
rowinterp = partial(fillmissing, cross_shore)
# apply to rows (along columns)
z = apply_along_axis(rowinterp, 1, z)

from enthought.mayavi import mlab
mlab.surf(z,mask=isnan(z), representation='wireframe')
mlab.show()

File: Openearthtool/python/applications/KMLDap/kmldap/templates/code.m
%% Run open earth tools settings so we have snctools loaded
oetsettings
url = '${config.get("jarkus.url")}'
nc_dump(url)
ids = nc_varget(url, 'id');
transect_index = find(ids == ${transect.id})-1;
%% Read all the years
year = nc_varget(url,'time')
%% Read all the altitude data for this transect
z = nc_varget(url,'altitude',[0,transect_index,0],[-1,1,-1]);
%% Read all the cross shore distances
cross_shore = nc_varget(url,'cross_shore');
%% Remove nan's
cross_shore_nonan = cross_shore(any(~isnan(z),1));
z_nonan = z(:,any(~isnan(z),1));
%% Create a plot.
surf(z_nonan)
shading flat
